src/lib.rs

```diff
@@ -2541,6 +2548,16 @@ impl Build {
             }
         }
 
+        if target.contains("msvc") && tool.family == ToolFamily::Gnu {
+            println!(
+                concat!(
+                    "cargo:warning=GNU compiler is not supported for this target, ",
+                    "consider --target={} instead."
+                ),
+                target.replace("msvc", "gnu")
```

Comment:
Literally "always"? No. I want to remind that [my actual position](https://github.com/rust-lang/cc-rs/issues/819#issuecomment-1640309753) is actually to fail the compilation or ignore the `$CC` environment value at this point. If the remark means that you agree, just say which option of the two is preferred, and I'll implement that. If not, then the question is if the suggestion is to soften the message to just "not supported" without making any suggestions?

Just in case for reference, the original rationale was that the `target.replace("msvc", "gnu")` arguably does the justice even if matching target does not exist. Again, the scenario is that the user actively sets `$CC` to gcc and asks to compile for msvc, which is error-prone effectively by definition. If the counterpart target doesn't exist, and the user feels strong about it, let him bring it up with Rust.

Reply:
Yes, I think it would be worse to suggest a target that does not exist. We either need to be sure it does exist or else just suggest a gnu target in less exact terms.

Reply:
Fixed.